src/Input/ChoiceParam.php

```diff
@@ -59,4 +57,28 @@ public function getQuestion(): Question
             $defaultValue
         );
     }
+
+    /**
+     * @param null|string|string[] $defaultValue
+     */
+    private function createDefaultPrompt($defaultValue): string
+    {
+        if (null === $defaultValue) {
+            return '';
+        }
+
+        if (is_array($defaultValue)) {
+            Assert::isList($defaultValue);
```

Comment:
The `isList()` assertion conflicts with the premise that this method accepts `string[]` (which is `array<int|string, string>`)

Reply:
I was under the impression from both the PHPDoc proposal and the Psalm documentation that `<type>[]` implied a list where all elements are of type `<type>`, and, as such, the keys are integers; is this not the case?

Reply:
Never mind; I should have typed this as `null|string|scalar[]` anyways, not with `string[]`.